test/integration/targets/collections_runtime_pythonpath/runme.sh

```diff
@@ -8,6 +8,7 @@ export PIP_DISABLE_PIP_VERSION_CHECK=1
 
 source virtualenv.sh
 
+pip install toml
```

Comment:
Why is this necessary? You shouldn't have to do this here.

Reply:
@webknjaz Actually we need `pip install packaging` too, in order for the test to run by itself, it seems.

Otherwise when I run

```
ansible-test integration collections_runtime_pythonpath --docker ubuntu2004 --python 3.8
```

I get:

```
=== Test that the module gets picked up if installed into site-packages ===
+ python -m pip.__main__ install pep517
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 193, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/usr/lib/python3/dist-packages/pip/__main__.py", line 16, in <module>
    from pip._internal.cli.main import main as _main  # isort:skip # noqa
  File "/usr/lib/python3/dist-packages/pip/_internal/cli/main.py", line 10, in <module>
    from pip._internal.cli.autocompletion import autocomplete
  File "/usr/lib/python3/dist-packages/pip/_internal/cli/autocompletion.py", line 9, in <module>
    from pip._internal.cli.main_parser import create_main_parser
  File "/usr/lib/python3/dist-packages/pip/_internal/cli/main_parser.py", line 7, in <module>
    from pip._internal.cli import cmdoptions
  File "/usr/lib/python3/dist-packages/pip/_internal/cli/cmdoptions.py", line 26, in <module>
    from pip._internal.models.format_control import FormatControl
  File "/usr/lib/python3/dist-packages/pip/_internal/models/format_control.py", line 4, in <module>
    from pip._vendor.packaging.utils import canonicalize_name
ModuleNotFoundError: No module named 'pip._vendor.packaging'
```

If I explicitly install that, the next error I get is:

```
=== Test that the module gets picked up if installed into site-packages ===
+ python -m pip.__main__ install pep517
Requirement already satisfied: pep517 in /root/ansible/test/results/.tmp/output_dir/venv/lib/python3.8/site-packages (0.8.2)
+ cp -r ansible-collection-python-dist-boo /root/ansible/test/results/.tmp/output_dir/
+ cd /root/ansible/test/results/.tmp/output_dir/ansible-collection-python-dist-boo
+ python -m pep517.build --binary --out-dir dist .
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 193, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/root/ansible/test/results/.tmp/output_dir/venv/lib/python3.8/site-packages/pep517/build.py", line 6, in <module>
    import toml
ModuleNotFoundError: No module named 'toml'
```

If I explicitly install both, the test passes.

Reply:
@webknjaz I'm not sure why this is necessary for 20.04 but not 18.04

Reply:
This might have ultiamtely been due to podman betraying me. Let me try reverting this change and see what CI does.

Reply:
@relrod looks like Ubuntu 20.04 ships broken pip then. `packaging` is vendored in pip upstream.

Reply:
And `toml` is a direct unconditional dependency of the `pep517` package. So when we `pip install` it below it should pull it in.

Reply:
> 01:15 + python -m pip.__main__ install pep517
01:15 Requirement already satisfied: pep517 in /root/ansible/test/results/.tmp/output_dir/venv/lib/python3.8/site-packages (0.8.2)

this is weird

Reply:
`source virtualenv.sh` is supposed to provide a clean venv and it seems like it's not that clean somehow.

Reply:
`virtualenv.sh` explicitly uses `--system-site-packages`

Reply:
Yeah, I see. But I was instructed by @mattclay to use it back when I was working on the PR. So this may mean that ubuntu-20 ships not completely installed dists in system site packages.

Reply:
There are several things going on here.

* As @sivel said we use `--system-site-packages`.
* Also in my local tests, `podman` betrayed me, and has an issue where `pip` randomly(!) can't install things: https://gist.githubusercontent.com/relrod/25ebdf7cbdd2b8247349cf180189551f/raw/365cf58441327a6957de76aa45676bba86adf926/gistfile1.txt -- this doesn't reproduce in Docker, just podman. So my previous (now deleted) comments didn't make sense.

I dug into why this is needed a bit more. Ubuntu 18.04's Python3 comes with virtualenv 15.x.x, and 20.04 comes with virtualenv 20.x.x, which has a newer version of setuptools embedded in it which enables pip517 stuff by default. When we enter the virtualenv and ask it to install `pep517`, it bails out because it thinks it has no reason to. But it still doesn't have toml which `python -m pep517.build --binary --out-dir dist .` needs to run successfully. So we do actually need `toml` installed here.

Reply:
> which has a newer version of setuptools embedded in it which enables pip517 stuff by default. When we enter the virtualenv and ask it to install pep517, it bails out because it thinks it has no reason to. But it still doesn't have toml which `python -m pep517.build --binary --out-dir dist .` needs to run successfully. So we do actually need `toml` installed here.

No, this is not entirely correct. Setuptools does not depend on `pep517` (the package, *not* the standard).

We shouldn't apply band-aid in places where some other bug manifests itself.

Do you have any log entries that show how `pep517` gets into that system site-packages env?

Reply:
> Do you have any log entries that show how pep517 gets into that system site-packages env?

Yes. Launch a fresh Ubuntu 20.04 container, install `python3` and `python3-virtualenv`.
Then in the container observe the following:

```
root@322e8c5b896e:/# python3.8 -m virtualenv --system-site-packages --python `which python3.8` venv
created virtual environment CPython3.8.2.final.0-64 in 539ms
  creator CPython3Posix(dest=/venv, clear=False, global=True)
  seeder FromAppData(download=False, pytoml=latest, colorama=latest, distlib=latest, html5lib=latest, contextlib2=latest, setuptools=latest, pkg_resources=latest, retrying=latest, urllib3=latest, msgpack=latest, chardet=latest, lockfile=latest, CacheControl=latest, pep517=latest, idna=latest, progress=latest, pyparsing=latest, six=latest, certifi=latest, requests=latest, distro=latest, wheel=latest, appdirs=latest, webencodings=latest, ipaddr=latest, packaging=latest, pip=latest, via=copy, app_data_dir=/root/.local/share/virtualenv/seed-app-data/v1.0.1.debian)
  activators BashActivator,CShellActivator,FishActivator,PowerShellActivator,PythonActivator,XonshActivator
root@322e8c5b896e:/# source venv/bin/activate
(venv) root@322e8c5b896e:/# pip install pep517
Requirement already satisfied: pep517 in /venv/lib/python3.8/site-packages (0.8.2)
```

This is from a fresh container.

Reply:
as in `docker run ubuntu:20.04`?

Reply:
Yeah. @mattclay and I dug into this a bunch tonight, there's a write-up here: #69203 and this PR is on hold for now until those issues are worked out.

Reply:
I think I'd add `pip check` right before creating the virtualenv in `virtualenv.sh` just to make sure that the promised "clean and ready to use" env is that indeed.